torch/_dynamo/symbolic_convert.py

```diff
@@ -3054,14 +3068,28 @@ def symbolic_locals_contain_module_class(self):
                 return True
         return False
 
+    def replace_tos_if_return_is_generator(self):
+        if (
+            len(self.stack)
+            and (tos := self.stack[-1])
+            and isinstance(tos, GeneratorObjectVariable)
+        ):
+            self.stack[-1] = ListIteratorVariable(
+                tos.force_unpack_var_sequence(self),
+                mutation_type=ValueMutationNew(),
+            )
+
     def _return(self, inst):
+        self.replace_tos_if_return_is_generator()
```

Comment:
Uh, does this work if we return a tuple(generator_object, generator_object)?

Reply:
Can we do the logic of turning the generator into a list in the reconstruct of the generatorobjectvariable instead?